From 4c3b552c6d3e8f08e0cd41577292103a5c549142 Mon Sep 17 00:00:00 2001
From: "Gary D. Gregory" <ggregory@apache.org>
Date: Tue, 19 Sep 2017 20:00:07 +0000
Subject: [PATCH] [VFS-291] ZIP archives are not properly closed after
 unzipping and cannot be deleted until the JVM exists. Do not close underlying
 ZipFile too aggressively; only do so if all open streams are not in use!

git-svn-id: https://svn.apache.org/repos/asf/commons/proper/vfs/trunk@1808941 13f79535-47bb-0310-9956-ffa450edef68
---
 .../org/apache/commons/vfs2/provider/zip/ZipFileObject.java  | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/zip/ZipFileObject.java b/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/zip/ZipFileObject.java
index b34252f5d6..69913019ec 100644
--- a/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/zip/ZipFileObject.java
+++ b/commons-vfs2/src/main/java/org/apache/commons/vfs2/provider/zip/ZipFileObject.java
@@ -153,6 +153,9 @@ protected void doAttach() throws Exception {
 
     @Override
     protected void doDetach() throws Exception {
-        getAbstractFileSystem().close();
+        ZipFileSystem afs = getAbstractFileSystem();
+        if (!afs.isOpen()) {
+            afs.close();
+        }
     }
 }
